From b19f05047d0949e144d006e91b1ef89e70eed7b2 Mon Sep 17 00:00:00 2001
From: Takeshi Kihara <takeshi.kihara.df@renesas.com>
Date: Thu, 2 Aug 2018 13:15:00 +0900
Subject: [PATCH 023/909] soc: renesas: Identify R-Car M3-W ES1.2

commit 92395ae82f5ba4bec18b7e3dc97177ae0a75da20 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

The Product Register of R-Car M3-W ES1.2 is the same SoC revision as
ES1.1. Leave the identifier of the revision as "ES1.1", and display
it as "ES1.1/ES1.2" only.

Signed-off-by: Takeshi Kihara <takeshi.kihara.df@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/soc/renesas/renesas-soc.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/soc/renesas/renesas-soc.c b/drivers/soc/renesas/renesas-soc.c
index d44d0e687ab8..5991422097a3 100644
--- a/drivers/soc/renesas/renesas-soc.c
+++ b/drivers/soc/renesas/renesas-soc.c
@@ -308,7 +308,9 @@ static int __init renesas_soc_init(void)
 						   product & 0xf);
 
 	pr_info("Detected Renesas %s %s %s\n", soc_dev_attr->family,
-		soc_dev_attr->soc_id, soc_dev_attr->revision ?: "");
+		soc_dev_attr->soc_id,
+		chipid && ((product & 0x7fff) == 0x5201) ? "ES1.1/ES1.2" :
+		soc_dev_attr->revision ?: "");
 
 	soc_dev = soc_device_register(soc_dev_attr);
 	if (IS_ERR(soc_dev)) {
-- 
2.17.1

